website/blog/2024-10-25-llama-index-workflow/index.md

```diff
@@ -0,0 +1,423 @@
+---
+title: Building Advanced RAG with MLflow and LlamaIndex Workflow
+description: A guide for using LlamaIndex Workflow with MLflow for building advanced QA application.
+slug: mlflow-llama-index-workflow
+authors: [yuki-watanabe]
+tags: [genai, mlops, mlflow-evaluate]
+---
+
+Augmenting LLMs with various data sources is a strong strategy to build LLM applications. However, as the system grows more complex, it is challenging to xxx and running quick development cycle for improvement.
+
+LlamaIndex Workflow is a great framework to build such compound system, and combined with MLflow brings efficiency and robustness in the development cycle, featuring debugging, tracking experiment, and evaluation for continuous improvement.
+
+In this blog, we will go through the journey of building an sophisticated chatbot with LlamaIndex Workflow and MLflow.
+
+## What is LlamaIndex Workflow?
+
+[LlamaIndex Workflow](https://docs.llamaindex.ai/en/stable/module_guides/workflow/) is an event-driven orchestration framework for designing dynamic AI applications. The core of LlamaIndex Workflow consists of:
+
+* `Steps` are units of execution, representing distinct actions in the workflow.
+
+* `Events` trigger these steps, acting as signals that control the workflow’s flow.
+
+* `Workflow` connects these two as a Python class. Each step is implemented as a method of the workflow class, defined with input and output events.
+
+This simple yet powerful abstraction allows you to break down complex tasks into manageable steps, enabling greater flexibility and scalability. As a nature of event-driven design, it is super easy to design parallel/asynchronous execution flow, which significantly enhances efficiency involve long-running tasks and provides production-ready scalability.
+
+## Why Use MLflow with LlamaIndex Workflow?
+
+Workflow provides great flexibility to design nearly arbitral execution flow. However, the great power comes with great responsibility. Without managing your change properly, it can become a chaos of states and configurations. A few weeks of later, "why did my workflow work?".
```

Comment:
```suggestion
Workflow provides great flexibility to design nearly arbitrary execution flows. However, with this great power comes a great responsibility. Without managing your change properly, it can become a chaotic mess of indeterminate states and confusing configurations. After a few dozen changes, you may be asking yourself, "how did my workflow even work?".
```